src/main/scala/streams/GameDef.scala

```diff
@@ -130,22 +130,24 @@ trait GameDef {
      * Returns the list of blocks that can be obtained by moving
      * the current block, together with the corresponding move.
      */
-    def neighbors: List[(Block, Move)] = ???
+    def neighbors: List[(Block, Move)] = {
+      List((left, Left), (right, Right), (up, Up), (down, Down))
+    }
 
     /**
      * Returns the list of positions reachable from the current block
      * which are inside the terrain.
      */
-    def legalNeighbors: List[(Block, Move)] = ???
+    def legalNeighbors: List[(Block, Move)] = neighbors.filter(pair => pair._1.isLegal)
 
     /**
      * Returns `true` if the block is standing.
      */
-    def isStanding: Boolean = ???
+    def isStanding: Boolean = b1.row == b2.row && b1.col == b2.col
```

Comment:
The default comparison of case classes is implemented by comparing the fields, and not by reference as normal classes. Therefore this implementation would be equal: 
```suggestion
    def isStanding: Boolean = b1 == b2
```
For more information see [here](https://stackoverflow.com/a/22103302/1018192)